packages/nimble-components/src/tree-view/index.ts

```diff
@@ -53,6 +62,45 @@ export class TreeView extends FoundationTreeView {
         return true;
     }
 
+    /**
+     * @internal
+     */
+    public clearTreeGroupSelection(): void {
+        const currentGroupSelection = this.querySelectorAll<TreeItem>(
+            `[${groupSelectedAttribute}]`
+        );
+        currentGroupSelection?.forEach(treeItem => treeItem.removeAttribute(groupSelectedAttribute));
+    }
+
+    /**
+     * @internal
+     */
+    public setGroupSelectionOnRootParentTreeItem(treeItem: HTMLElement): void {
+        this.clearTreeGroupSelection();
+
+        let currentItem: HTMLElement | null | undefined = treeItem;
+        while (
+            currentItem?.parentElement !== this
+            && currentItem?.parentElement !== null
+        ) {
+            currentItem = currentItem?.parentElement;
+        }
+
+        if (currentItem) {
+            currentItem.setAttribute(groupSelectedAttribute, 'true');
+        }
+    }
+
+    // This prevents the toggling of selected state when a TreeItem is clicked multiple times,
+    // which is what the FAST TreeItem allows
+    private onSelectedChange(event: Event): void {
+        const toggledElement = event.target as ISelectable;
+        // only process selection
+        if (toggledElement.selected) {
```

Comment:
Programmatically I can can the selection to false for a selected element such than none are selected anymore but the group selected indicator is still shown

Reply:
Had to put in a reference counting scheme to know when to remove the group selected indicator.